src/models/artefact.ts

```diff
@@ -1,40 +1,67 @@
 import { Polygon } from '@/utils/Polygons';
+import { ImagedObject } from './imaged-object';
+import { ArtefactDTO } from '@/dtos/artefact';
+import { Side } from './misc';
+
 
 export class Artefact {
-    public id: number;
-    public positionId: number;
-    public shapeId: number;
-    public scrollVersionId: number;
-    public name: string;
-    public mask: Polygon | undefined;
-    public transformMatrix: any; // TODO: Change to matrix type?
-    public rect: string;
-    public imageCatalogId: number; // Probably not needed
-    public sqeImageId: number;  // Probable not needed
-
-    constructor(obj: any) {
-        if (obj.side !== 0) {
-            console.error('Received a non-recto artefact ', obj);
-            throw new Error('Non-recto artefacts are not supported');
+    // public static createNew(editionId: number, imagedObject: ImagedObject, name: string, id: number) {
+    //      // editionId: EditionInfo
+    //     const artefact = new Artefact({
+    //         id,
+    //         editionId, // editionId.versionId,
+    //         imagedObjectId: imagedObject.id,
+    //         name,
+    //         mask: new Polygon(''),
+    //         transformMatrix: '',
+    //         zOrder: 1,
+    //         side: 'recto',
+    //     } as Artefact);
+
+    //     if (!imagedObject.artefacts) {
+    //         imagedObject.artefacts = [];
+    //     }
+    //     imagedObject.artefacts.push(artefact);
+
+    //     return artefact;
+    // }
+
+    // Default values specified to remove an error - we initialize them in the constructor or in copyFrom.
+    // Typescript does not approve of that and shows an error, because it doesn't analyze copyFrom.
+    public id = 0;
+    public editionId = 0;
+    public imagedObjectId = '';
+    public name = '';
+    public mask = {} as Polygon;
+    public transformMatrix = undefined as any; // TODO: Change to matrix type?
```

Comment:
We will not move the matrix into the Polygon class, as the Polygon class represents a Polygon. We will add a Matrix class (which for now will just be a simple wrapper around a string, with no parsing) and add a Mask class which will contain a Matrix and a Polygon. This will require some wide spread changes, but they are trivial to make.